notebooks/cytomining-ecosystem/visualize-landscape-fall-2023.py

```diff
@@ -0,0 +1,688 @@
+        in an early state of their user base (both in age and stars),
+        relative to other projects.
+        """,
+        "section": "User base",
+    }
+)
+
+fig_usage_stars.show()
+
+# +
+# Create an icicle chart using Plotly Express
+df_treemap = df_projects.copy()
+df_treemap["GitHub Stars"] = np.where(
+    df_treemap["GitHub Stars"] == 0, np.nan, df_treemap["GitHub Stars"]
+)
+fig_usage_stars_treemap = px.treemap(
+    df_treemap.sort_values(by="category"),
+    title="GitHub Stars Project Tree Map (click to zoom)",
+    path=["category", "Project Name"],
+    values="GitHub Stars",
+    color="GitHub Stars",
+    color_continuous_scale="Viridis",
+    width=1200,
+    height=700,
+)
+
+fig_collection.append(
+    {
+        "plot": fig_usage_stars_treemap,
+        "description": """
+        This plot demonstrates the magnitude of total GitHub stars by project and category.
+        Click the category boxes to zoom into focus on only those projects.
+        """,
+        "findings": """
+        Here we can see relevant open source far outsize other categories.
+        Within the microscopy analysis tools category we can observe that Napari has many more stars than others.
+        Projects of <code>loi-focus</code> are nearly invisible in comparison to other projects with
+        much greater age.
+        """,
+        "section": "User base",
+    }
+)
+
+# Show the icicle chart
+fig_usage_stars_treemap.show()
+
+# +
+# scatter plot for maturity based on project
+
+# gather the number of lines of code
+df_projects["total lines of GitHub detected code"] = (
+    df_projects["GitHub Detected Languages"]
+    .dropna()
+    .apply(lambda x: sum(value if value is not None else 0 for value in x.values()))
+)
+
+# add log of github stars to help visualize
+df_projects["total lines of GitHub detected code (Log Scale)"] = np.log(
+    df_projects["total lines of GitHub detected code"].apply(
+        # move 0's to None to avoid divide by 0
+        lambda x: x
+        if x > 0
+        else None
+    )
+)
+
+
+fig_maturity_loc_and_age = px.scatter(
+    df_projects.sort_values(by="category"),
+    hover_name="Project Name",
+    x="Duration Created to Now in Years",
+    y="total lines of GitHub detected code (Log Scale)",
+    width=1200,
+    height=500,
+    color="category",
+    color_discrete_sequence=category_color_sequence,
+)
+
+
+# customize the chart layout
+fig_maturity_loc_and_age.update_layout(
+    title=f"Project Lines of Code and Age in Years (click categories for further focus)",
+    yaxis_title="Total Lines of Code (Log Scale)",
+    xaxis_title="Project Age (years)",
+)
+
+fig_collection.append(
+    {
+        "plot": fig_maturity_loc_and_age,
+        "description": """
+        This plot shows the lines of code (LoC) and age in years to help indicate relative project maturity.
+        Lines of code in this case are the sum total of GitHub detected programming language dialects.
```

Comment:
```suggestion
        Lines of code in this case are the sum total of all detected programming languages in the latest GitHub repository.
```

Accurate? I tried to simplify/clarify

Reply:
Thanks - yes this is accurate, I'll add this in!